chain/network/src/peer/peer_actor.rs

```diff
@@ -392,13 +392,14 @@ impl PeerActor {
         self.send_message_with_encoding(msg, Encoding::Borsh);
     }
 
+    #[tracing::instrument(
+        level = "trace",
+        target = "network",
+        "send_message_with_encoding",
+        skip_all,
+        fields(msg_type = msg.msg_variant())
+    )]
```

Comment:
Can you give me a quick TLDR of what it does?

Reply:
This is pretty much a different way to write `tracing::span!` that covers the entire function and happens correctly handle `async fn` by default. I updated the style guide with some information about it today: https://near.github.io/nearcore/practices/style.html#spans